_posts/2016-04-04-code-smells-part-II.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+name: code-smells-part-one
+title: Code Smells – Part II
+date: 2016-04-04 12:10:00 +00:00
+author: Ana Nogal
+canonical:
+    name: my personal blog
+    href: http://www.ananogal.com/blog/code-smells-part-two/
+image:
+    src: /assets/img/custom/blog/code-smells_II.jpg
+    attribution:
+        text: Via Flickr/Creative Commons/Brian Fitzgerald (Creative Commons - Attribution-ShareAlike 2.0 Generic license)
+        href: https://www.flickr.com/photos/brian-fitzgerald/3334353375
+tags:
+- object-orientation abusers
+- change preventers
+- code smells
+- refactoring
+
+---
+In the last post, [Code Smells - Part I](http://codurance.com/2016/03/17/code-smells-part-I),  I talked about the bloaters: they are code smells that can be identified as Long Methods, Large Classes, Primitive Obsessions, Long Parameter List and Data Clumps. In this one, I would like to dig into the **Object-Orientation Abusers** and the **Change Preventers**. 
+In today's post, I'll talk about a few refactoring techniques, but there are a lot more. You can find a good reference to all of then in [Refactoring.com](http://refactoring.com/catalog/) by Martin Fowler.
+
+## Object-Orientation Abusers
+
+This type of code smell usually happens when the object-oriented principles are incomplete or incorrectly applied.
+
+
+#### Switch Statements
+
+This case is simple to identify: we have a switch case. But you should consider a smell too if you find a sequence of ifs. (that's a switch case in disguise).
+Why switch statements are bad? Because when a new condition is added, you have to find all the switch code and modify it. 
```

Comment:
You have to find every occurrence of that switch case

Why are there several occurrences?

What if I encapsulate the switch into a method, is it acceptable then?